synapse/storage/room.py

```diff
@@ -187,6 +192,30 @@ def _store_room_name_txn(self, txn, event):
                 }
             )
 
+            self._store_event_search_txn(
+                txn, event, "content.name", event.content["name"]
+            )
+
+    def _store_room_message_txn(self, txn, event):
+        if hasattr(event, "content") and "body" in event.content:
+            self._store_event_search_txn(
+                txn, event, "content.body", event.content["body"]
+            )
+
+    def _store_event_search_txn(self, txn, event, key, value):
```

Comment:
Would it make sense to move this into a method on the db engine rather than doing "isinstance" checks?